### PR TITLE
AsyncBenchmark: avoid doing work on leader I/O thread

### PR DESCRIPTION
**Pull request by pruivo**

### CODE REVIEW

**Review by franz1981 — CHANGES_REQUESTED** [review]
LGTM i suggest using a `LInkedTransferQueue`

File: src/org/jgroups/perf/AsyncCounterBenchmark.java [hunk 1]
```diff
@@ -81,15 +87,26 @@ private void updateTime(long timeNanos) {
...
-            final long currentTime = System.nanoTime();
-            updateTime(currentTime - start);
-            return updateCounter(counter, currentTime);
+    private void updateCounter(AsyncCounter counter) {
+        requests.increment();
+        long start = System.nanoTime();
+        counter.addAndGet(deltaSupplier.getAsLong()).thenRun(() -> {
+            long time = System.nanoTime();
+            updateTime(time - start);
+            queue.offer(time);
```
Comment (franz1981) [inline src/org/jgroups/perf/AsyncCounterBenchmark.java:96]: TBH capturing latency shouldn't be a big deal...I suggest switching to using a non thread-safe HdrHistogram (in the other PR) so, given that I/O is happening on a single thread, there is no need for atomics or any synchronization mechanism

**Review by franz1981 — CHANGES_REQUESTED** [review]

**Comment by franz1981** [comment]
Thinking about it twice (as suggested on https://github.com/belaban/jgroups-raft/pull/171#discussion_r865653893) there's no need to use any queue and just handle histogram update on the I/O thread (with non thread safe constructs), but we still need a mechanism to count and react on completions.

This can use a mechanism similar to the one I've shown in another PR or submitting a continuation in the same thread (in the form of 'executor::execute(this::submit)`.
Ideally the version I did, although ugly, was meant to save submitting a continuation for each completion, if the updater thread was still running (that's very beneficial with bursty scenario)

**Comment by pruivo** [comment]
I'm closing this one since I'm unable to reproduce the issue anymore (async is always faster in hdd and in tmpfs in the latest master commit)

**Comment by pruivo** [comment]
ps. although, we probably need to come up with a fixed-rate perf tool.

**Pull request closed without merging**